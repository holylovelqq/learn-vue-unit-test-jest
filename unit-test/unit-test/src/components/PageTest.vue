<template>
  <div>
    <p v-for="item in items" :key="item">{{ item }}</p>
  </div>
</template>
<script>
export default {
  name: "page-test",
  // props: ["items"],
  data() {
    return {
      items: ["first", "second"]
    };
  }
};
</script>
<style scoped>
</style>
