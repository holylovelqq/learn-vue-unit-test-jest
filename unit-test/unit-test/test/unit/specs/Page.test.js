// page.test.js
import Vue from 'vue'
import PageTest from '@/components/PageTest'

describe('page-test', () => {
  it('renders p', () => {
    const Constructor = Vue.extend(PageTest)
    const vm = new Constructor().$mount()
    expect(vm.$el.querySelectorAll('p')).toHaveLength(2)
    expect(vm.$el.querySelectorAll('p')[0].textContent).toEqual('first')
  })
  it('matchs snapshot', () => {
    const Constructor = Vue.extend(PageTest)
    const vm = new Constructor().$mount()
    expect(vm.$el).toMatchSnapshot()
  })
})
