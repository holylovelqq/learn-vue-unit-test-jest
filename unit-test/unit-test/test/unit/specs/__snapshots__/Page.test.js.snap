// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page-test matchs snapshot 1`] = `
<div>
  <p>
    first
  </p>
  <p>
    second
  </p>
</div>
`;
